packages/airnode-utilities/src/evm/gas-prices/gas-oracle.ts

```diff
@@ -0,0 +1,296 @@
+      referencePercentileGasPrice.percentileGasPrice,
+      maxDeviationMultiplier
+    );
+
+    // Return the percentile for the latest block if within the limit
+    if (isWithinDeviationLimit) {
+      return { type: 0, gasPrice: latestPercentileGasPrice.percentileGasPrice };
+    }
+
+    throw new Error(
+      `Latest block percentileGasPrice exceeds the max deviation multiplier limit set to (${maxDeviationMultiplier}%).`
+    );
+  } else {
+    throw new Error(`Unable to get enough blocks or transactions to calculate percentileGasPrice.`);
+  }
+};
+
+export const attemptGasOracleStrategy = async (
+  provider: Provider,
+  gasOracleConfig: config.GasPriceOracleStrategy
+): Promise<GasTarget> => {
+  switch (gasOracleConfig.gasPriceStrategy) {
+    case 'latestBlockPercentileGasPrice':
+      return await fetchLatestBlockPercentileGasPrice(provider, gasOracleConfig);
+    case 'providerRecommendedGasPrice':
+      return await fetchProviderRecommendedGasPrice(provider, gasOracleConfig);
+    case 'providerRecommendedEip1559GasPrice':
+      return await fetchProviderRecommendedEip1559GasPrice(provider, gasOracleConfig);
+    default:
+      throw new Error('Unsupported gas price oracle strategy.');
+  }
+};
+
+// Get gas price based on gas price oracle strategies
+export const getGasPrice = async (
+  provider: Provider,
+  chainOptions: config.ChainOptions
+): Promise<LogsData<GasTarget>> => {
+  const { gasPriceOracle, fulfillmentGasLimit } = chainOptions;
+  const startTime = Date.now();
+
+  const goProcessGasPriceOracleStrategies = await go(
+    () => processGasPriceOracleStrategies(provider, gasPriceOracle, startTime),
+    { totalTimeoutMs: GAS_ORACLE_STRATEGY_MAX_TIMEOUT_MS }
+  );
+
+  // Return the constant strategy gas price if all other strategies fail
+  if (!goProcessGasPriceOracleStrategies.success || !goProcessGasPriceOracleStrategies.data[1]) {
+    const constantGasPriceConfig = gasPriceOracle.find(
+      (strategy) => strategy.gasPriceStrategy === 'constantGasPrice'
+    ) as config.ConstantGasPriceStrategy;
+    const constantGasTarget = fetchConstantGasPrice(constantGasPriceConfig);
+
+    const log = logger.pend(
+      'INFO',
+      `All oracle strategies failed to return a gas price, returning constant gas price set to ${ethers.utils.formatUnits(
+        constantGasTarget.gasPrice,
+        'gwei'
+      )} gwei.`
+    );
+    const strategyLogs = goProcessGasPriceOracleStrategies.success ? goProcessGasPriceOracleStrategies.data[0] : [];
+
+    return [[log, ...strategyLogs], getGasTargetWithGasLimit(constantGasTarget, fulfillmentGasLimit)];
+  }
+
+  const [logs, strategyGasTarget] = goProcessGasPriceOracleStrategies.data;
+
+  return [logs, getGasTargetWithGasLimit(strategyGasTarget, fulfillmentGasLimit)];
+};
+
+export const processGasPriceOracleStrategies = async (
+  provider: Provider,
+  gasPriceOracleConfig: config.GasPriceOracleConfig,
+  startTime: number
```

Comment:
You actually don't need to pass `startTime` from `getGasPrice`. It's not used there anywhere, you might as well instantiate it here in `processGasPriceOracleStrategies`.